dj/construction/build_planning.py

```diff
@@ -137,20 +140,27 @@ async def optimize_level_by_cost(build_plan: BuildPlan) -> Tuple[int, Database]:
         if combined_level:
             try:
                 cheapest_levels.append(
-                    (i, await get_cheapest_online_database(combined_level)),
+                    (
+                        i,
+                        await get_cheapest_online_database(
+                            combined_level,
+                            check_database_online=check_database_online,
+                        ),
+                    ),
                 )
                 some_db = True
             except Exception as exc:  # pylint: disable=broad-except    # pragma: no cover
                 if "No active database found" not in str(exc):
                     raise exc
 
-    if not some_db:  # pragma: no cover
+    if check_database_online and not some_db:  # pragma: no cover
```

Comment:
Seeing this flag getting passed around to eventually be used here, I'm wondering if this `some_db` could just be returned at the end and the check can happen at the end instead of deep within this flow of logic. In other words, if this is updated so that `some_db` is blindly passed along, then you would only need the `check_database_online` argument for the `dj.api.helpers.get_query` function and you can move this check out to that function:

*psuedo-code, the dj.api.helpers.get_query function*
```py
async def get_query(........, check_database_online=False)
    ...
    query_ast, optimal_database, some_db = await build_node_for_database(...)
    if check_database_online and not some_db:
        raise Exception("No database found that can execute this query.")
```